src/service/exception_handlers.py

```diff
@@ -0,0 +1,70 @@
+"""
+Exception handlers for the FastAPI application.
+"""
+
+import logging
+
+from fastapi import HTTPException, Request, status
+from fastapi.responses import JSONResponse
+
+from src.service.error_mapping import map_error
+from src.service.exceptions import SparkManagerError
+
+logger = logging.getLogger(__name__)
+
+
+def _format_error(
+    status_code: int, error_code: int, error_type_str: str, message: str | None
+):
+    """Format error response with consistent structure."""
+    return JSONResponse(
+        status_code=status_code,
+        content={
+            "error": error_code,
+            "error_type": error_type_str,
+            "message": message or error_type_str,
+        },
+    )
+
+
+async def universal_error_handler(request: Request, exc: Exception):
+    """
+    Universal handler for all types of exceptions.
+
+    Handles:
+    - SparkManagerError and its subclasses (including all specific error types)
+    - HTTPException from FastAPI
+    - Generic exceptions
+    """
+    # Default values
+    error_code = 0
+    error_type_str = "Unknown error"
+    message = None
+    status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+
+    if isinstance(exc, SparkManagerError):
+        # handle SparkManagerError and subclasses
+        error_type, status_code = map_error(exc)
+
+        # Extract values from error_type if available
+        if error_type:
+            error_code = error_type.error_code
+            error_type_str = error_type.error_type
+        else:
+            error_type_str = type(exc).__name__
```

Comment:
You're mixing semantics for the error type string here and below

Reply:
can you make suggestions? I am fine with it as long as it provides some information about the error. If you have a strong option on it, I am fine whatever your suggestions are.

Reply:
To me there are 2 separate things here

* application specific error codes - those have entries in the error map and tell the user there's something wrong with what they did and give a specific integer code and human readable name for what that is. They're supposed to help users fix their application specific problems, and users should be able to look them up somewhere in the documentation (at some point) and get help
* http error codes - these are the standard http response status strings

Those shouldn't be mixed in the same field

Reply:
Yea sure. Can you make suggestions on what should error_type_str  be assigned to? I am fine with the current way or what you think is appropriate.

Reply:
There's a couple options I can think of

* For the http errors or errors without an error mapping, just don't populate that field and let the user get the http error from the response
* Add a dedicated field for http errors

Reply:
Could you make github code suggestions regarding to the specific part of the code you're referring to? (e.g. what `error_type_str = type(exc).__name__` should be in your option).  That way, I can address it/accept it directly and get the review through faster. As the repo owner, I’d prefer not to make assumptions about what you're referring to, especially if it’s something I may not consider crucial.